catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/ModalRegistry.view.js

```diff
@@ -15,264 +15,292 @@
+                this.rebind(properties);
+            },
+            submitData: function () {
+                wreqr.vent.trigger('beforesave');
+                var view = this;
+                var service = this.getConfig();
+                if (service) {
+                    service.save().then(function () {
                             wreqr.vent.trigger('refreshRegistries');
                             view.closeAndUnbind();
-                        }
-                    },
-                    function () {
-                        wreqr.vent.trigger('refreshRegistries');
-                    }).always(function () {
+                        },
+                        function () {
+                            wreqr.vent.trigger('refreshRegistries');
+                        }).always(function () {
                         view.closeAndUnbind();
                     });
-            }
-        },
-        onClose: function() {
-            this.modelBinder.unbind();
-            this.$el.off('hidden.bs.modal');
-            this.$el.off('shown.bs.modal');
-        },
-        closeAndUnbind: function() {
-            this.modelBinder.unbind();
-            this.$el.modal("hide");
-        },
-        rebind: function (properties) {
-            var $boundData = this.$el.find('.bound-controls');
-            var bindings = Backbone.ModelBinder.createDefaultBindings($boundData, 'name');
-            delete bindings.value;
-            this.modelBinder.bind(properties, $boundData, bindings);
-        },
-        initRadioButtonsUI: function(boundModel) {
-            var $radios = this.$el.find('input[type=radio]');
-            var view = this;
-
-            _.each($radios, function(radio) {
-                var $radio = view.$(radio);
-                var $label = $radio.closest('label.btn');
+                }
+            },
+            urlChanged: function (evt) {
+                var url = evt.target.value;
+                var hostname;
+                if (url) {
+                    hostname = this.extractHostnameFromUrl(url);
+                }
+                if (hostname && this.urlIsValid(url)) {
+                    var oldName;
+                    var nameToSet = hostname;
+                    var toCheck = "(" + hostname + ")";
+                    if (!this.type || this.type === "CSW Registry Store") {
+                        nameToSet += " (CSW)";
+                        toCheck += " (CSW)";
+                    } else {
+                        nameToSet += " (" + this.type + ")";
+                        toCheck += " (" + this.type + ")";
+                    }
+                    var count = 0;
+                    this.registry.get('collection').models.find(function (registry) {
+                        if (registry.id) {
+                            if (registry.id.endsWith(nameToSet) || registry.id.endsWith(nameToSet + ")") || registry.id.endsWith(toCheck) || registry.id.endsWith(toCheck + ")")) {
```

Comment:
It feels like we only need one variable here nameToSet or toCheck. Since we are creating the name we should only need to check one case.